plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityCannotEnterHive.java

```diff
@@ -2,97 +2,41 @@
 
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.DurationTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.tags.Attribute;
 import org.bukkit.entity.Bee;
-import org.bukkit.entity.Entity;
 
-public class EntityCannotEnterHive implements Property {
+public class EntityCannotEnterHive extends EntityProperty<DurationTag> {
 
-    public static boolean describes(ObjectTag entity) {
-        if (!(entity instanceof EntityTag)) {
-            return false;
-        }
-        Entity bukkitEntity = ((EntityTag) entity).getBukkitEntity();
-        return bukkitEntity instanceof Bee;
-    }
-
-    public static EntityCannotEnterHive getFrom(ObjectTag entity) {
-        if (!describes(entity)) {
-            return null;
-        }
-        else {
-            return new EntityCannotEnterHive((EntityTag) entity);
-        }
-    }
-
-    public static final String[] handledTags = new String[] {
-            "cannot_enter_hive"
-    };
-
-    public static final String[] handledMechs = new String[] {
-            "cannot_enter_hive"
-    };
-
-    public EntityCannotEnterHive(EntityTag entity) {
-        this.entity = entity;
-    }
-
-    EntityTag entity;
+    // <--[property]
+    // @object EntityTag
+    // @name cannot_enter_hive
+    // @input DurationTag
+    // @description
+    // Controls the minimum duration until a Bee is able to enter a hive.
+    // -->
 
-    public Bee getBee() {
-        return (Bee) entity.getBukkitEntity();
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Bee;
     }
 
     @Override
-    public String getPropertyString() {
-        return new DurationTag((long) getBee().getCannotEnterHiveTicks()).identify();
+    public DurationTag getPropertyValue() {
+        return new DurationTag(as(Bee.class).getCannotEnterHiveTicks());

```

Comment:
I don't think removing the `(long)` conversion is valid here. Have you tested that this property still works as intended, including returning values that make sense?